src/main/java/net/minecraftforge/common/IAnvilRecipe.java

```diff
@@ -0,0 +1,64 @@
+package net.minecraftforge.common;
+
+import java.util.ArrayList;
+import net.minecraft.item.ItemStack;
+import net.minecraft.world.World;
+
+public interface IAnvilRecipe
+{
+    /**
+     * Returns the input for this recipe, any mod accessing this value should never
+     * manipulate the values in this array as it will effect the recipe itself.
+     *
+     * @return The recipes input vales. Each value can be an ItemStack, List<ItemStack> or null.
+     */
+    ArrayList<Object> getInput();
```

Comment:
I suggest to use `List<ItemStack> getInputLeft` and `List<ItemStack> getInputRight` methods, as `ArrayList` may not be a good return value, and single item stacks can be turned into a singleton list.